src/ngraph/runtime/cpu/mkldnn_emitter.hpp

```diff
@@ -1400,8 +1159,575 @@ namespace ngraph
+                                                               *mkldnn_primitives[results_idx]);
+                    }
+
+                    mkldnn_primitives[conv_idx] = prim;
+                }
+
+                template <bool with_bias>
+                void build_inner_product_forward(std::vector<mkldnn::primitive*>& mkldnn_primitives,
+                                                 const mkldnn::inner_product_forward::desc& desc,
+                                                 const mkldnn::primitive_attr& attr,
+                                                 const mkldnn::engine& engine,
+                                                 const std::vector<size_t>& deps,
+                                                 size_t ip_idx)
+                {
+                    size_t input_idx, weights_idx, results_idx, bias_idx;
+                    input_idx = deps[0];
+                    weights_idx = deps[1];
+                    mkldnn_primitives[input_idx] =
+                        new mkldnn::memory({{desc.data.src_desc}, engine}, nullptr);
+                    mkldnn_primitives[weights_idx] =
+                        new mkldnn::memory({{desc.data.weights_desc}, engine}, nullptr);
+                    if (with_bias)
+                    {
+                        bias_idx = deps[2];
+                        results_idx = deps[3];
+                        mkldnn_primitives[bias_idx] =
+                            new mkldnn::memory({{desc.data.bias_desc}, engine}, nullptr);
+                    }
+                    else
+                    {
+                        results_idx = deps[2];
+                    }
+                    mkldnn_primitives[results_idx] =
+                        new mkldnn::memory({{desc.data.dst_desc}, engine}, nullptr);
+
+                    mkldnn::primitive* prim;
+                    if (with_bias)
+                    {
+                        prim = new mkldnn::inner_product_forward({desc, attr, engine},
+                                                                 *mkldnn_primitives[input_idx],
+                                                                 *mkldnn_primitives[weights_idx],
+                                                                 *mkldnn_primitives[bias_idx],
+                                                                 *mkldnn_primitives[results_idx]);
+                    }
+                    else
+                    {
+                        prim = new mkldnn::inner_product_forward({desc, attr, engine},
+                                                                 *mkldnn_primitives[input_idx],
+                                                                 *mkldnn_primitives[weights_idx],
+                                                                 *mkldnn_primitives[results_idx]);
+                    }
+
+                    mkldnn_primitives[ip_idx] = prim;
+                }
+#endif
```

Comment:
This is actually a lot of code replication :). Would it be possible to use an approach similar to FORMAT macro in some cases to reduce part of the replication?